renku/core/commands/update.py

```diff
@@ -46,96 +52,211 @@ def update_workflows():
+        inputs = [i.actual_value for p in plans for i in p.inputs]
+        client.pull_paths_from_storage(*inputs)
 
     delete_indirect_files_list(client.path)
 
-    # Execute the workflow and relocate all output files.
-    # FIXME get new output paths for edited tools
-    # output_paths = {path for _, path in workflow.iter_output_files()}
-    execute(path, output_paths=output_paths)
+    started_at_time = local_now()
 
-    paths = [o.produces.path for o in workflow.outputs]
+    modified_outputs = _execute_workflow_helper(plans=plans, client=client)
 
-    add_to_git(client.repo.git, *paths)
+    ended_at_time = local_now()
 
-    if client.repo.is_dirty():
-        commit_msg = f"renku {command_name}: committing {len(paths)} newly added files"
+    add_to_git(client.repo.git, *modified_outputs)
 
+    if client.repo.is_dirty():
+        postfix = "s" if len(modified_outputs) > 1 else ""
+        commit_msg = f"renku {command_name}: committing {len(modified_outputs)} modified file{postfix}"
         committer = Actor(f"renku {__version__}", version_url)
-
         client.repo.index.commit(commit_msg, committer=committer, skip_hooks=True)
 
-    workflow_name = f"{uuid.uuid4().hex}_{command_name}.yaml"
-
-    path = client.workflow_path / workflow_name
-
-    workflow.update_id_and_label_from_commit_path(client, client.repo.head.commit, path)
-
-    # TODO: implement properly with new database
-    # if not workflow.subprocesses:  # Update parameters if there is only one step
-    #     _update_run_parameters(run=workflow, working_dir=client.path)
-
-    # cls = WorkflowRun if workflow.subprocesses else ProcessRun
-    # activity = cls.from_run(run=workflow, path=path, update_commits=update_commits)
-    # activity.to_yaml(path=path)
-    # client.add_to_activity_index(activity)
-
-    # activity_gateway.add(activity)
-
-
-# def _update_run_parameters(run, working_dir):
-
-#     default_parameters = {p.name: p for p in run.run_parameters}
-
-#     indirect_parameters = read_indirect_parameters(working_dir)
-# for name, value in indirect_parameters.items():
-#     id_ = RunParameter.generate_id(run_id=run._id, name=name)
-#     parameter = RunParameter(id=id_, name=name, value=value)
-#     default_parameters[name] = parameter
-
-# run.run_parameters = list(default_parameters.values())
+    database = database_dispatcher.current_database
+
+    activities = []
+
+    for plan in plans:
+        # NOTE: Update plans are copies of Plan objects. We need to use the original Plan objects to avoid duplicates.
+        original_plan = database["plans"].get(plan.id)
+        activity = Activity.from_plan(plan=original_plan, started_at_time=started_at_time, ended_at_time=ended_at_time)
+        activity_gateway.add(activity)
+        activities.append(activity)
+
+    if len(activities) > 1:
+        activity_collection = ActivityCollection(activities=activities)
+        activity_gateway.add_activity_collection(activity_collection)
```

Comment:
It looks like the `"activity-collections"` only exists to enable some tests. it is a bit odd to me to store something in users' repositories that is only used for our tests, essentially littering the database with unneeded data.

I guess we could have a `TestingActivityGateway(ActivityGateway)` in `tests/` and inject that for tests, to only have this index when testing code?

Reply:
`ActivityCollection` is to mark that these activities have been executed together as a result of an update or a rerun. So, it's not just for testing. I was not sure what other metadata we need to include here (specifically if we need a link in the `Activity` to its `ActivityCollection` if any). What do you think?

Reply:
a link from ActivityCollection to Activity makes sense to me. Best to discuss it with the KG team as well